commit 530f735bdfe1b6baffd1ae96b663c294d32dfcf0 (refs/changes/61/3361/1)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2018-02-13T14:04:59-08:00 (1 year, 8 months ago)
    
    TRITON-130 mounting non-existing volume with same name from more than one VM fails

diff --git a/lib/workflows/nfs-volumes/index.js b/lib/workflows/nfs-volumes/index.js
index f443c65..962784b 100644
--- a/lib/workflows/nfs-volumes/index.js
+++ b/lib/workflows/nfs-volumes/index.js
@@ -292,7 +292,10 @@ function provisionNfsVolumes(job, cb) {
                             name: volumeName,
                             owner_uuid: job.params.owner_uuid,
                             predicate: JSON.stringify({
-                                eq: ['state', 'ready']
+                                or: [
+                                    {eq: ['state', 'ready']},
+                                    {eq: ['state', 'creating']}
+                                ]
                             })
                         }, function onListVolumes(listVolumesErr, volumes) {
                             var loadedVolume;
